### ISSUE TITLE
Invalid type NeosWikiAssetGenerator.Type_Processors.NeosTypeProcessor[]

### ISSUE THREAD

**Issue by RobertBaruch** [Aerizeon/NeosWikiAssetGenerator#3]
Seems like the plugin doesn't work with the latest version of Neos:

```
System.Exception: Error instantiating type: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeLoadException: Invalid type NeosWikiAssetGenerator.Type_Processors.NeosTypeProcessor[] for instance field System.Collections.Generic.List`1[NeosWikiAssetGenerator.Type_Processors.NeosTypeProcessor]:_items
  at (wrapper managed-to-native) System.Reflection.MonoCMethod.InternalInvoke(System.Reflection.MonoCMethod,object,object[],System.Exception&)
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00002] in <9577ac7a62ef43179789031239ba8798>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00014] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic) [0x000a8] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00009] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00027] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00020] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.WorkerManager.Instantiate (System.Type type) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0  ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeLoadException: Invalid type NeosWikiAssetGenerator.Type_Processors.NeosTypeProcessor[] for instance field System.Collections.Generic.List`1[NeosWikiAssetGenerator.Type_Processors.NeosTypeProcessor]:_items
  at (wrapper managed-to-native) System.Reflection.MonoCMethod.InternalInvoke(System.Reflection.MonoCMethod,object,object[],System.Exception&)
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00002] in <9577ac7a62ef43179789031239ba8798>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.MonoCMethod.InternalInvoke (System.Object obj, System.Object[] parameters) [0x00014] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic) [0x000a8] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00009] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Threading.StackCrawlMark& stackMark) [0x00027] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00020] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <9577ac7a62ef43179789031239ba8798>:0 
  at FrooxEngine.WorkerManager.Instantiate (System.Type type) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0 
   --- End of inner exception stack trace ---
  at FrooxEngine.WorkerManager.Instantiate (System.Type type) [0x00020] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.ContainerWorker`1[C].AttachComponent (System.Type type, System.Boolean runOnAttachBehavior, System.Action`1[T] beforeAttach) [0x00007] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.ComponentAttacher.OnAddComponentPressed (FrooxEngine.IButton button, FrooxEngine.ButtonEventData eventData, System.String typename) [0x00024] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.ButtonRelay`1[T].OnPressed (FrooxEngine.IButton button, FrooxEngine.ButtonEventData eventData) [0x0001d] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.ButtonRelayBase.Pressed (FrooxEngine.IButton button, FrooxEngine.ButtonEventData eventData) [0x00062] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Button+<>c__DisplayClass65_0.<RunPressed>b__0 (FrooxEngine.IButtonPressReceiver r) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.ContainerWorker`1[C].ForeachComponent[T] (System.Action`1[T] callback, System.Func`2[T,TResult] callbackStopper, System.Boolean cacheItems, System.Boolean excludeDisabled) [0x00034] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.ContainerWorker`1[C].ForeachComponent[T] (System.Action`1[T] callback, System.Boolean cacheItems, System.Boolean exludeDisabled) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Button.RunPressed (FrooxEngine.ButtonEventData eventData) [0x00067] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Button.OnPressBegin (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00035] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.InteractionElement.ProcessEvent (FrooxEngine.UIX.Canvas+InteractionData eventData) [0x00235] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Canvas.ProcessTouchEvent (FrooxEngine.TouchEventInfo& eventInfo, System.Collections.Generic.List`1[T] filters) [0x00398] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Canvas.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.UIX.Canvas.FrooxEngine.ITouchable.OnTouch (FrooxEngine.TouchEventInfo& eventInfo) [0x00000] in <c491a41aea9a42ddbe10865c040e96bd>:0 
  at FrooxEngine.TouchSource.SendTouchEvent (FrooxEngine.ITouchable touchable, FrooxEngine.TouchEventInfo& touchInfo) [0x00018] in <c491a41aea9a42ddbe10865c040e96bd>:0 

   at void BaseX.UniLog.Error(string message, bool stackTrace)
   at void FrooxEngine.DebugManager.Error(object obj, bool stackTrace)
   at void FrooxEngine.TouchSource.SendTouchEvent(ITouchable touchable, in TouchEventInfo touchInfo)
   at void FrooxEngine.TouchSource.UpdateCurrentTouchable(ITouchable touchable, in float3 point, in float3 direction, in float3 directHitPoint, bool touch)
   at void FrooxEngine.TouchSource.UpdateTouch()
   at void FrooxEngine.InteractionLaser.UpdateLaser(Nullable<float> maintainDistance, Nullable<float> overrideSmoothing, Nullable<InteractionOrigin> overrideOrigin, bool targettingOnly)
   at void FrooxEngine.CommonTool.UpdateInteractionAndLaser(bool targettingOnly)
   at void FrooxEngine.CommonTool.OnInputEvaluate()
   at void FrooxEngine.InputGroup.Update(float deltaTime)
   at void FrooxEngine.InputBindingManager.Update()
   at bool FrooxEngine.World.RefreshStep()
   at bool FrooxEngine.World.Refresh()
   at void FrooxEngine.WorldManager.UpdateStep(double maxMilliseconds)
   at bool FrooxEngine.WorldManager.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngine.Engine.UpdateStep(double maxMilliseconds)
   at void FrooxEngine.Engine.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngineRunner.Update()
```

**Comment by RobertBaruch** [comment]
Seems like List<NeosTypeProcessor> needs to be List<INeosTypeProcessor>.

**Comment by Aerizeon** [comment]
Implemented suggested fix in 4bf69f8c9d289eebf23e21bbb9d91827200a2ded

Thanks!
